src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -324,6 +324,13 @@ public static SyntaxList<AttributeListSyntax> GetAttributeLists(this SyntaxNode?
                 current.Parent is not ConditionalAccessExpressionSyntax)
             {
                 current = current.Parent;
+
+                // Optional exclamations might follow the conditional operation. For example:
+                // a.b?.$$c!!!!()
+                while (current.IsKind(SyntaxKind.SuppressNullableWarningExpression))
```

Comment:
move up into the IsKind check on line 318?

Reply:
Sure